monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,324 @@
+            # Get the bounds of the buffered box, this is the one that will will be fed to the remote ID display provider to query            
+            self.query_bboxes.append(QueryBoundingBox(name=box_diagonals[box_id]['name'], shape=buffered_box, time_step_after = box_diagonals[box_id]['time_step_after'], time_step_before = box_diagonals[box_id]['time_step_before']))
+            
+
+    def generate_flight_grid(self):
+
+        ''' Generate a series of boxes within the bounding box to have areas for different flights '''
+        # Compute the box where the flights will be created. For a the sample bounds given, over Bern, Switzerland, a division by 2 produces a cell_size of 0.0025212764739985793, a division of 3 is 0.0016808509826657196 and division by 4 0.0012606382369992897. As the cell size goes smaller more number of flights can be accomodated within the grid. For the study area bounds we build a 3x2 box for six flights by creating 3 column 2 row grid. 
+        N_COLS = 3
+        N_ROWS = 2
+        cell_size_x = (self.maxx - self.minx)/(N_COLS) # create three columns
+        cell_size_y = (self.maxy - self.miny)/(N_ROWS) # create two rows
+        grid_cells = [] 
+        for u0 in range(0, N_COLS):  # 3 columns           
+            x0 =  self.minx + (u0 * cell_size_x)            
+            for v0 in range(0,N_ROWS): # 2 rows
+                y0 = self.miny + (v0 *cell_size_y)
+                x1 = x0 + cell_size_x
+                y1 = y0 + cell_size_y
+                grid_cells.append(shapely.geometry.box(x0, y0, x1, y1))                
+        
+        self.flight_grid = grid_cells
+        
+
+    def generate_flight_paths_points(self):
+
+        ''' For each of the boxes allocated to the operator, get the centroid and buffer to get a flight path. A 70 m radius is provided to have flight paths within each of the boxes '''
+
+        # Iterate over the flight_grid
+        for grid_cell in self.flight_grid:
+            center = grid_cell.centroid
+            ## Transfrom to buffer 140 m diameter circle on which the drone will fly            
+            transformer = Transformer.from_crs("epsg:4326", "epsg:3857")
+            transformed_x,transformed_y = transformer.transform(center.x, center.y)
+            pt = Point(transformed_x,transformed_y)
+            buffer = pt.buffer(70) # build a buffer so that the radius is 70m for the track
+            buffer_points =  zip(buffer.exterior.coords.xy[0], buffer.exterior.coords.xy[1])
+            proj_buffer_points = []
+            # reproject back to ESPG 4326
+            transformer2 = Transformer.from_crs("epsg:3857", "epsg:4326")
+            for point in buffer_points:
+                x = point[0]
+                y = point[1]
+                x, y = transformer2.transform(x, y)
+                proj_buffer_points.append((x, y))
+            buffered_path = Polygon(proj_buffer_points)
+            altitude = 70.0
```

Comment:
Hmm, since we are adding the altitude to the transformed co-ordinates, my thinking that they will be in the WGS1984 co-ordinate reference system. The transformation is done in L151. I think this should work.

Reply:
On L164, it looks like the `alt` of each `FlightPoint` is set directly to `altitude` -- where would the altitude conversion occur?  (L151 is now an initialization instruction)

Reply:
Just to understand this, to convert the altitude you need some understanding of the ground level, I can check what is the ground level near FOCA offices, are you suggesting taking this approach?

Reply:
There are a few different reasonable ways for the simulator to behave, but the end result must always be meters above the WGS84 ellipsoid (because that's what's in the standard).  Requiring the user to specify the WGS84 altitude of ground level is a pretty easy approach for now.

Reply:
OK. I will add a additional field to have the user enter WGS84 altitude of ground level and we can derive flight altitude.